src/DependencyInjection/SonataBlockExtension.php

```diff
@@ -69,6 +69,11 @@ public function load(array $configs, ContainerBuilder $container): void
         $loader->load('exception.xml');
         $loader->load('commands.xml');
 
+        // NEXT_MAJOR: Remove this check and the legacy_core.xml file.
+        if (class_exists(\Sonata\Cache\CacheManagerInterface::class)) {
```

Comment:
I added this in order to keep the old behavior by overriding the service if the user installed SonataCache.